bolt-sidecar/src/state/execution.rs

```diff
@@ -115,9 +124,21 @@ impl<C: StateFetcher> ExecutionState<C> {
     /// If the commitment is valid, it will be added to the block template and its account state
     /// will be cached. If this is succesful, any callers can be sure that the commitment is valid
     /// and SHOULD sign it and respond to the requester.
-    pub async fn try_commit(&mut self, request: &CommitmentRequest) -> Result<(), ValidationError> {
+    pub async fn check_commitment_validity(
+        &mut self,
+        request: &CommitmentRequest,
+    ) -> Result<Address, ValidationError> {
         let CommitmentRequest::Inclusion(req) = request;
 
+        // Check if there is room for more commitments
+        if let Some(template) = self.get_block_template(req.slot) {
+            if template.transactions_len() >= self.max_commitments_per_slot.get() {
+                return Err(ValidationError::MaxCommitmentsReachedForSlot(
+                    self.max_commitments_per_slot.get(),
+                ));
+            }
```

Comment:
From the `MaxCommitmentsReachedForSlot` error description below, it seems like we should use `req.slot` here instead, or am I misinterpreting it?

```rust
    #[error("Max commitments reached for slot {0}")]
    MaxCommitmentsReachedForSlot(usize),
```

Reply:
Good catch, adding both the slot and the maximum number of constraints reached